lesson11/index.js

```diff
@@ -0,0 +1,68 @@
+const jsonUrl = 'https://myjson.dit.upm.es/api/bins/77w5';
+
+const ul = document.getElementById('js-list');
+
+const createLoading = () => {
+    const li = document.createElement("li");
+    const loadingImg = document.createElement('img');
+
+    li.style.listStyle = "none";
+    li.id = "js-loading";
+    loadingImg.src = 'loading-circle.gif';
+    ul.appendChild(li).appendChild(loadingImg);
+};
+
+const removeLoading = () => document.getElementById('js-loading').remove();
+
+const addList = (val) => {
+    const fragment = document.createDocumentFragment();
+    for (let i = 0; i < val.length; i++) {
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        const img = document.createElement('img');
+    
+        a.href = `${val[i].a}.html`;
+        a.textContent = `${val[i].text}`;
+        img.src = val[i].img;
+        img.alt = val[i].alt;
+    
+        fragment.appendChild(li).appendChild(a).appendChild(img);
+    }
+
+    ul.appendChild(fragment);
+};
+
+/**
+ * Promiseを実行し、resolveの値を返す関数
+ * @return {promise object} 
+ */
+ const fetchData = (json) => {
+    return new Promise( resolve => {
+        setTimeout(() => {
+            resolve(json);
+        }, 3000);
+    });
+};
```

Comment:
IMO
My interpretation was that the fetchData() would now be changed to json fetch details.

```suggestion
const fetchData = async () => {
  try {
    const response = await fetch(jsonUrl); // 中身はresponseオブジェクトを含むpromise
    const json = await response.json(); //response オブジェクトから JSON を抽出
    return json.data; //追加
  } catch (error) {
    console.error(error);
  }
};

```

Reply:
Changed the process in the function for the function to get json.
[f1dc2cd](https://github.com/AyakaOnozuka131/JavaScript-lessons/pull/11/commits/f1dc2cdee13c3fc63263d038e713be160ef8a83b)